components/core/src/clp/ffi/ir_stream/Deserializer.hpp

```diff
@@ -0,0 +1,67 @@
+#ifndef CLP_FFI_IR_STREAM_DESERIALIZER_HPP
+#define CLP_FFI_IR_STREAM_DESERIALIZER_HPP
+
+#include <memory>
+
+#include <outcome/single-header/outcome.hpp>
+
+#include "../../ReaderInterface.hpp"
+#include "../../time_types.hpp"
+#include "../KeyValuePairLogEvent.hpp"
+#include "../SchemaTree.hpp"
+
+namespace clp::ffi::ir_stream {
+/**
+ * A class for deserializing log events from a CLP IR stream of kv-pair IR format.
+ *
+ * This class:
+ * - maintains all the necessary internal data structure to track deserialization state;
+ * - provide APIs to deserialize log events from an IR stream.
```

Comment:
I think this is obvious, right?

Reply:
Right. I've added more details to explain the APIs should work as a transaction